infrastructure/terraform/modules/activation/main.tf

```diff
@@ -410,26 +437,110 @@ data "external" "ga4_measurement_properties" {
+# This key can then be used for various encryption operations, 
+# such as encrypting data before storing it in Google Cloud Storage 
+# or protecting secrets within your application.
+resource "google_kms_crypto_key" "crypto_key_regional" {
+  name     = "crypto-key-${random_id.random_suffix.hex}"
+  key_ring = google_kms_key_ring.key_ring_regional.id
+}
+
+# Defines an IAM policy that explicitly grants the Secret Manager service account 
+# the ability to encrypt and decrypt data using a specific CryptoKey. This is a 
+# common pattern for securely managing secrets, allowing Secret Manager to encrypt 
+# or decrypt data without requiring direct access to the underlying encryption key material.
+data "google_iam_policy" "crypto_key_encrypter_decrypter" {
+  binding {
+    role = "roles/cloudkms.cryptoKeyEncrypterDecrypter"
+
+    members = [
+      "serviceAccount:${google_project_service_identity.secretmanager_sa.email}"
+    ]
+  }
+
+  depends_on = [
+    google_project_service_identity.secretmanager_sa,
+    google_kms_key_ring.key_ring_regional,
+    google_kms_crypto_key.crypto_key_regional
+  ]
+}
+
+# It sets the IAM policy for a KMS CryptoKey, specifically granting permissions defined 
+# in another data source.
+resource "google_kms_crypto_key_iam_policy" "crypto_key" {
+  crypto_key_id = google_kms_crypto_key.crypto_key_regional.id
+  policy_data = data.google_iam_policy.crypto_key_encrypter_decrypter.policy_data
+}
+
+# It sets the IAM policy for a KMS Key Ring, granting specific permissions defined 
+# in a data source.
+resource "google_kms_key_ring_iam_policy" "key_ring" {
+  key_ring_id = google_kms_key_ring.key_ring_regional.id
+  policy_data = data.google_iam_policy.crypto_key_encrypter_decrypter.policy_data
+}
+
 # This module stores the values ga4-measurement-id and ga4-measurement-secret in Google Cloud Secret Manager.
 module "secret_manager" {
   source     = "GoogleCloudPlatform/secret-manager/google"
-  version    = "0.4.0"
-  project_id = null_resource.check_secretmanager_api.id != "" ? module.project_services.project_id : var.project_id
+  version    = "~> 0.1"
+  project_id = google_kms_crypto_key_iam_policy.crypto_key.etag != "" && google_kms_key_ring_iam_policy.key_ring.etag != "" ? module.project_services.project_id : var.project_id
   secrets = [
     {
       name                  = "ga4-measurement-id"
       secret_data           = (var.ga4_measurement_id == null || var.ga4_measurement_secret == null) ? data.external.ga4_measurement_properties[0].result["measurement_id"] : var.ga4_measurement_id
-      automatic_replication = true
+      #automatic_replication = true
     },
     {
       name                  = "ga4-measurement-secret"
       secret_data           = (var.ga4_measurement_id == null || var.ga4_measurement_secret == null) ? data.external.ga4_measurement_properties[0].result["measurement_secret"] : var.ga4_measurement_secret
-      automatic_replication = true
+      #automatic_replication = true
```

Comment:
Remove the comment, or set the falg to false

Reply:
done